src/components/ServicesEntry/services-entry.jsx

```diff
@@ -0,0 +1,128 @@
+import Button from 'terra-button';
+import Dialog from 'terra-dialog';
+import Spacer from 'terra-spacer';
+import Text from 'terra-text';
+
+import styles from './services-entry.css';
+import BaseEntryBody from '../BaseEntryBody/base-entry-body';
+import retrieveDiscoveryServices from '../../retrieve-data-helpers/discovery-services-retrieval';
+
+const propTypes = {
+  isOpen: PropTypes.bool,
+  closePrompt: PropTypes.func,
+};
+
+export class ServicesEntry extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      isOpen: this.props.isOpen,
+      userInput: '',
+      shouldDisplayError: false,
+      errorMessage: '',
+    };
+
+    this.handleCloseModal = this.handleCloseModal.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.isOpen !== nextProps.isOpen) {
+      this.setState({ isOpen: nextProps.isOpen });
+    }
+  }
+
+  handleCloseModal() {
+    this.setState({ isOpen: false, shouldDisplayError: false, errorMessage: '' });
+    if (this.props.closePrompt) { this.props.closePrompt(); }
+  }
+
+  handleChange(e) {
+    this.setState({ userInput: e.target.value });
+  }
+
+  async handleSubmit() {
+    if (this.state.userInput === '' || !this.state.userInput || !this.state.userInput.trim()) {
+      this.setState({ shouldDisplayError: true, errorMessage: 'Enter a valid discovery endpoint' });
+    }
+    let checkUrl = this.state.userInput.trim();
+    if (!/^(https?:)?\/\//i.test(checkUrl)) {
+      checkUrl = `http://${checkUrl}`;
+      this.setState({
+        userInput: checkUrl,
+      });
+    }
+    try {
+      await retrieveDiscoveryServices(checkUrl).then(() => {
+        this.handleCloseModal();
+      });
+    } catch (e) {
+      this.setState({
+        shouldDisplayError: true,
+        errorMessage: 'Failed to connect to the discovery endpoint. See console for details.',
+      });
+    }
+  }
+
+  render() {
+    const headerContainer = (
+      <Text weight={700} fontSize={20}>Add CDS Services</Text>
+    );
+
+    const footerContainer = (
+      <div className={styles['right-align']}>
+        <Button text="Save" variant="emphasis" onClick={this.handleSubmit} />
+        <Spacer marginLeft="small" isInlineBlock>
+          <Button text="Cancel" onClick={this.handleCloseModal} />
+        </Spacer>
+      </div>
+    );
+
+    return (
+      <div>
+        <Modal
+          ariaLabel="CDS Services"
+          isOpen={this.state.isOpen}
+          closeOnEsc
+          closeOnOutsideClick
+          onRequestClose={this.handleCloseModal}
+          classNameModal={styles['fixed-size']}
+        >
+          <Dialog
+            header={headerContainer}
+            footer={footerContainer}
+            onClose={this.handleCloseModal}
+          >
+            <BaseEntryBody
+              formFieldLabel="Enter Discovery Endpoint URL"
```

Comment:
How about ```Enter discovery endpoint url``` ? Unless its a requirement to have it all Capitalized on the first character.